heat/core/statistics.py

```diff
@@ -1328,32 +1293,54 @@ def reduce_vars_elementwise(output_shape_i):
                 mu_in = 0.0
 
             n = x.lnumel
-            var_tot = factories.zeros((x.comm.size, 3))
-            var_proc = factories.zeros((x.comm.size, 3))
-            var_proc[x.comm.rank][0] = var_in
-            var_proc[x.comm.rank][1] = mu_in
-            var_proc[x.comm.rank][2] = float(n)
+            var_tot = factories.zeros((x.comm.size, 3), device=x.device)
+            var_proc = factories.zeros((x.comm.size, 3), device=x.device)
+            var_proc[x.comm.rank] = var_in, mu_in, float(n)
             x.comm.Allreduce(var_proc, var_tot, MPI.SUM)
 
             for i in range(1, x.comm.size):
-                merged = merge_vars(
-                    var_tot[0, 0],
-                    var_tot[0, 1],
-                    var_tot[0, 2],
-                    var_tot[i, 0],
-                    var_tot[i, 1],
-                    var_tot[i, 2],
+                var_tot[0, 0], var_tot[0, 1], var_tot[0, 2] = merge_moments(
+                    (var_tot[0, 0], var_tot[0, 1], var_tot[0, 2]),
+                    (var_tot[i, 0], var_tot[i, 1], var_tot[i, 2]),
+                    bessel=bessel,
                 )
-                var_tot[0, 0] = merged[0]
-                var_tot[0, 1] = merged[1]
-                var_tot[0, 2] = merged[2]
-
             return var_tot[0][0]
 
     else:  # axis is given
```

Comment:
Small inconsistency: When axis is set, integer tensors are not supported any longer. Numpy does support ints and floats, while pytorch only supports floats in general. This behaviour is also in 'mean'.

Reply:
because torch does not support ints so we dont either. if its needed, the easiest way to do this would be to cast the input to a float. I dont think that we need to do this though. i dont think that we need to be so religious to numpy